src/ToolBox/ToolAnalyzer.php

```diff
@@ -5,12 +5,17 @@
 namespace PhpLlm\LlmChain\ToolBox;
 
 use PhpLlm\LlmChain\Exception\InvalidToolImplementation;
+use PhpLlm\LlmChain\StructuredOutput\SchemaFactory;
 
 final readonly class ToolAnalyzer
 {
+    private SchemaFactory $schemaFactory;
+
     public function __construct(
         private ParameterAnalyzer $parameterAnalyzer = new ParameterAnalyzer(),
+        ?SchemaFactory $schemaFactory = null,
```

Comment:
We would need the `SchemaFactory` here now, because of resolving the ResponseFormat before creating the `Metadata` object, good to go? Concerns?

Reply:
that's the reality if we want to combine those features, true :+1: